src/hdmf/container.py

```diff
@@ -847,18 +824,7 @@ def extend(self, arg):
 
         :param arg: The iterable to add to the end of this VectorData
         """
-        if self.term_set is None:
-            self.__data = extend_data(self.__data, arg)
-        else:
-            bad_data = []
-            for item in arg:
-                try:
-                    self.append(item)
-                except ValueError:
-                    bad_data.append(item)
-            if len(bad_data)!=0:
-                msg = ('"%s" is not in the term set.' % ', '.join([str(item) for item in bad_data]))
-                raise ValueError(msg)
```

Comment:
I think we should keep this validation as well for the same reasons mentioned above.

Reply:
I don't think we would need this here. This was moved to the wrapper. The logic remains the same. Now it calls the wrapper append, which is essentially this.